java/src/com/github/bazel_contrib/contrib_rules_jvm/javaparser/generators/ClasspathParser.java

```diff
@@ -284,23 +295,39 @@ public Void visitVariable(VariableTree node, Void unused) {
       return super.visitVariable(node, unused);
     }
 
-    private void checkFullyQualifiedType(Tree identifier) {
+    @Nullable
+    private Set<String> checkFullyQualifiedType(Tree identifier) {
+      Set<String> types = new TreeSet<>();
       if (identifier.getKind() == Tree.Kind.IDENTIFIER
           || identifier.getKind() == Tree.Kind.MEMBER_SELECT) {
         String typeName = identifier.toString();
-        if (typeName.contains(".")) {
+        List<String> components = Splitter.on(".").splitToList(typeName);
+        if (currentFileImports.containsKey(components.get(0))) {
+          String importedType = currentFileImports.get(components.get(0));
+          usedTypes.add(importedType);
+          types.add(importedType);
+        } else if (components.size() > 1) {
           usedTypes.add(typeName);
+          types.add(typeName);
+        } else {
+          String expandedImport = currentFileImports.get(typeName);
+          if (expandedImport != null) {
+            usedTypes.add(expandedImport);
+            types.add(expandedImport);
+          }
         }
       } else if (identifier.getKind() == Tree.Kind.PARAMETERIZED_TYPE) {
         Tree baseType = ((ParameterizedTypeTree) identifier).getType();
         checkFullyQualifiedType(baseType);
         ((ParameterizedTypeTree) identifier)
-            .getTypeArguments()
-            .forEach(this::checkFullyQualifiedType);
+            .getTypeArguments().stream()
+                .flatMap(t -> checkFullyQualifiedType(t).stream())
+                .forEach(types::add);
```

Comment:
TIL member functions support the `::` syntax :D 
This is fine code, but something in me just assumed that `types::add` would be a static function. Maybe it was the `::`.